### PR TITLE
Dshank/noise aware labels

### PR DESCRIPTION
**Pull request by Concomitant**
First pass making classifier support dictionaries with floats as values(e.g. probability a probability distribution with labels)

### CODE REVIEW

File: finetune/target_models/classifier.py [hunk 1]
```diff
@@ -36,11 +36,32 @@ def _target_encoder(self):
     def feed_shape_type_def(self):
         return super(ClassificationPipeline, self).feed_shape_type_def()
 
+class NoisyClassificationPipeline(BasePipeline):
+    def resampling(self, Xs, Y, context=None):
+        if context is not None:
```
Comment (madisonmay) [inline finetune/target_models/classifier.py]: Nuke this

Comment (madisonmay) [reply finetune/target_models/classifier.py]: Or better -- raise a warning if oversample is set

File: finetune/encoding/target_encoders.py [hunk 2]
```diff
@@ -77,6 +78,28 @@ def inverse_transform(self, one_hot):
...
         return ys
 
+class NoisyLabelEncoder(LabelEncoder, BaseEncoder):
+
+    # Overriding the fit method...
+    # Fit method may not be necessary at all if pandas is
+    # consistent about how it chooses columns
+    # TODO: Check
+    def fit(self, y):
+        self.classes_ = list(pd.DataFrame(y[:1]).columns)
```
Comment (madisonmay) [inline finetune/encoding/target_encoders.py]: Assumption: y[0] contains all class labels

File: finetune/target_models/classifier.py [hunk 3]
```diff
@@ -51,6 +72,169 @@ def __init__(self, **kwargs):
...
+
+        return processed
+
+    def _predict_op(self, logits, **kwargs):
+        return tf.contrib.seq2seq.hardmax(logits)
+
+    def _predict_proba_op(self, logits, **kwargs):
+        return tf.nn.softmax(logits, -1)
+
+class NoisyClassifier(BaseModel):
```
Comment (madisonmay) [inline finetune/target_models/classifier.py]: Inherit from classifier and just override required methods

File: finetune/encoding/target_encoders.py (hunk 2 above)
Comment (Concomitant) [reply finetune/encoding/target_encoders.py]: Yeah that is an assumption. I was prematurely optimizing--didn't want to do the whole pd.DataFrame conversion just to get labels. Now that I think about it, normal encoders do need to read through all the labels to fit anyway... Haven't profiled this yet though so all speculation.

File: finetune/target_models/classifier.py (hunk 3 above)
Comment (Concomitant) [reply finetune/target_models/classifier.py]: You got it!
